src/limit_order.cpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+#include "..\include\TradingEngine\limit_order.h"
+
+#include <memory>
+using std::shared_ptr;
+
+#include <unordered_set>
+using std::unordered_set;
+
+#include "..\include\TradingEngine\transactions.h" 
+#include "..\include\TradingEngine\order_type.h"
+#include "..\include\TradingEngine\order_status.h"
+#include "..\include\TradingEngine\order.h"
+
+const shared_ptr<Order>& LimitOrder::getHead() {
+	return limit_order_head;
+}
+
+void LimitOrder::setHead(const std::shared_ptr<Order>& incoming_order) {
+	limit_order_head = incoming_order;
+}
+
+const shared_ptr<Order>& LimitOrder::getTail() {
+	return limit_order_tail;
+}
+
+void LimitOrder::setTail(const std::shared_ptr<Order>& incoming_order) {
+	limit_order_tail = incoming_order;
+}
+
+int LimitOrder::getListLength() const {
+	return limit_order_length;
+}
+
+int LimitOrder::updateListLength(const int& update_increment) {
+	return limit_order_length += update_increment;
+}
+
+void LimitOrder::addOrder(Order& incoming_order) {
+
+	/* Create pointer to incoming order */
+	const std::shared_ptr<Order>& incoming_order_ptr = std::make_shared<Order>(incoming_order);
+
+	/* Increment LinkedList length. */
+	updateListLength(1);
+
+	/* First Order */
+	if (getListLength() == 1) {
```

Comment:
you shouldn't rely on getters and setters in the cpp of the class itself, you do literally have access to private members, you should access them directly :D Setters and getters are for the users of the objects.